prow/flagutil/github.go

```diff
@@ -331,6 +332,31 @@ func (o *GitHubOptions) GitHubClientWithAccessToken(token string) (github.Client
 	return client, err
 }
 
+func (o *GitHubOptions) GitClientFactory(cookieFilePath string, cacheDir *string, dryRun bool) (gitv2.ClientFactory, error) {
+	var gitClientFactory gitv2.ClientFactory
+	if cookieFilePath != "" && o.TokenPath == "" && o.AppPrivateKeyPath == "" {
+		opts := gitv2.ClientFactoryOpts{
+			CookieFilePath: cookieFilePath,
+		}
+		if cacheDir != nil && *cacheDir != "" {
+			opts.CacheDirBase = cacheDir
+		}
+		var err error
+		gitClientFactory, err = gitv2.NewClientFactory(opts.Apply)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create git client from cookieFile: %v", err)
```

Comment:
Isn't the cookieFile only for Gerrit?

Reply:
initially I thought cookieFile is a generic git thing. Did a quick google search now and realized that it's only for gerrit. So you're right that this shouldn't belong to github flag. Any suggestions where this log should belong to?

Reply:
Because we want to avoid refactoring too much as you pointed out offline, I'm OK with keeping it here with a TODO to move it out (and maybe update the error message to something like `failed to create ... (cookeFile is only for Gerrit)`.

Reply:
ack, done.